sif/internal/sif_util.go

```diff
@@ -0,0 +1,238 @@
+// +build linux
+
+package internal
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/containers/image/v5/sif/sif"
+	"github.com/pkg/errors"
+
+	imgspecv1 "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+type SifImage struct {
+	fimg      sif.FileImage
+	rootfs    *sif.Descriptor
+	deffile   *sif.Descriptor
+	defReader *io.SectionReader
+	cmdlist   []string
+	runscript *bytes.Buffer
+	env       *sif.Descriptor
+	envReader *io.SectionReader
+	envlist   []string
+}
+
+func LoadSIFImage(path string) (image SifImage, err error) {
+	// open up the SIF file and get its header
+	image.fimg, err = sif.LoadContainer(path, true)
+	if err != nil {
+		return
+	}
+
+	// check for a system partition and save it
+	image.rootfs, _, err = image.fimg.GetPartPrimSys()
+	if err != nil {
+		return SifImage{}, errors.Wrap(err, "looking up rootfs from SIF file")
+	}
+
+	// look for a definition file object
+	searchDesc := sif.Descriptor{Datatype: sif.DataDeffile}
+	resultDescs, _, err := image.fimg.GetFromDescr(searchDesc)
+	if err == nil && resultDescs != nil {
+		// we assume in practice that typical SIF files don't hold multiple deffiles
+		image.deffile = resultDescs[0]
+		image.defReader = io.NewSectionReader(image.fimg.Fp, image.deffile.Fileoff, image.deffile.Filelen)
```

Comment:
`defReader` is only used in `generateConfig` immediately below, so maybe it can just be a local variable.